src/ui/public/utils/aggressive_parse.js

```diff
@@ -0,0 +1,27 @@
+import { isString, startsWith } from 'lodash';
+
+/**
+ * Serializes the given object into a JSON string
+ *
+ * All properties that begin with $ throughout the entire object are omitted.
+ * If a custom JSON serializer function is passed, then the given object is
+ * passed through it before being re-stringified with the native stringify.
+ *
+ * The space argument is passed unaltered to the native stringify.
+ */
+export function toJson(object, jsonFn, space) {
```

Comment:
Yep.  I had originally implemented `omit$` and `deepOmit$` functions that `object` would pass through before being dropped into `angular.json` directly, but I hit the infinite recursion wall when testing on some random objects throughout kibana, so I scrapped them in favor of relying on the native construct to catch that sort of thing.